From a97ceaafbc26846f0273f066803f3e76097493a6 Mon Sep 17 00:00:00 2001
From: Brett Mastbergen <bmastbergen@ciq.com>
Date: Thu, 11 Jun 2026 14:30:59 -0400
Subject: [PATCH] arm64: errata: Mitigate TLBI errata on Microsoft Azure Cobalt
 100 CPU

jira VULN-187519
cve CVE-2025-10263
commit-author Will Deacon <will@kernel.org>
commit -
commit-source-sha 1940e70a8144bf75e6df26bf6f600862ea7f7ea1
commit-source arm64
upstream-diff silicon-errata.rst at different path and required manual
  conflict resolution due to missing entries in our branch.
  arm64_repeat_tlbi_cpus[] is a flat struct midr_range array on 4.18;
  added MIDR_ALL_VERSIONS() entry directly instead of nested
  ERRATA_MIDR_RANGE_LIST().

Commit fb091ff39479 ("arm64: Subscribe Microsoft Azure Cobalt 100 to ARM
Neoverse N2 errata") states that Microsoft Azure Cobalt 100 CPU "is a
Microsoft implemented CPU based on r0p0 of the ARM Neoverse N2 CPU, and
therefore suffers from all the same errata.".

So enable the workaround for the latest broadcast TLB invalidation bug
on these parts.

	Signed-off-by: Will Deacon <will@kernel.org>
(cherry picked from commit 1940e70a8144bf75e6df26bf6f600862ea7f7ea1)
	Signed-off-by: Brett Mastbergen <bmastbergen@ciq.com>
---
 Documentation/arm64/silicon-errata.rst | 3 +++
 arch/arm64/Kconfig                     | 1 +
 arch/arm64/kernel/cpu_errata.c         | 1 +
 3 files changed, 5 insertions(+)

diff --git a/Documentation/arm64/silicon-errata.rst b/Documentation/arm64/silicon-errata.rst
index d7eb3d18bdc2..55d88df1e5b4 100644
--- a/Documentation/arm64/silicon-errata.rst
+++ b/Documentation/arm64/silicon-errata.rst
@@ -192,3 +192,6 @@ stable kernels.
 +----------------+-----------------+-----------------+-----------------------------+
 | Fujitsu        | A64FX           | E#010001        | FUJITSU_ERRATUM_010001      |
 +----------------+-----------------+-----------------+-----------------------------+
++----------------+-----------------+-----------------+-----------------------------+
+| Microsoft      | Azure Cobalt 100| #4193789        | ARM64_ERRATUM_4118414       |
++----------------+-----------------+-----------------+-----------------------------+
diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 58a10be7164b..0c9dc4086c70 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -650,6 +650,7 @@ config ARM64_ERRATUM_4118414
 	  * ARM Neoverse-V2 erratum 4193787
 	  * ARM Neoverse-V3 erratum 4193784
 	  * ARM Neoverse-V3AE erratum 4193784
+	  * Microsoft Azure Cobalt 100 4193789
 	  * NVIDIA Olympus erratum T410-OLY-1029
 
 	  On affected cores, some memory accesses might not be completed by
diff --git a/arch/arm64/kernel/cpu_errata.c b/arch/arm64/kernel/cpu_errata.c
index 02c6898e5606..5817a5958d8d 100644
--- a/arch/arm64/kernel/cpu_errata.c
+++ b/arch/arm64/kernel/cpu_errata.c
@@ -686,6 +686,7 @@ static const struct midr_range arm64_repeat_tlbi_cpus[] = {
 	MIDR_ALL_VERSIONS(MIDR_NEOVERSE_V3),
 	MIDR_ALL_VERSIONS(MIDR_NEOVERSE_V3AE),
 	MIDR_ALL_VERSIONS(MIDR_NVIDIA_OLYMPUS),
+	MIDR_ALL_VERSIONS(MIDR_MICROSOFT_AZURE_COBALT_100),
 #endif
 	{},
 };
-- 
2.47.3

